test/testvalueflow.cpp

```diff
@@ -3202,6 +3203,38 @@ class TestValueFlow : public TestFixture {
         ASSERT_EQUALS(true, values.front().isPossible() || values.back().isPossible());
         ASSERT_EQUALS(true, values.front().intvalue == 0 || values.back().intvalue == 0);
     }
+
+    void valueFlowTerminatingCond() {
+        const char* code;
+
+        code = "void f(int i, int j) {\n"
+               "    if (i > j) return;\n"
+               "    bool x = i > j;\n"
+               "    bool b = x;\n"
+               "}\n";
+        ASSERT_EQUALS(true, testValueOfX(code, 4U, 0));
+
+        code = "void f(int i, int j) {\n"
+               "    if (i > j) return;\n"
+               "    bool x = i < j;\n"
+               "    bool b = x;\n"
+               "}\n";
+        ASSERT_EQUALS(true, testValueOfX(code, 4U, 1));
+
+        code = "void f(int i) {\n"
+               "    if (i > 0) return;\n"
+               "    bool x = i > 0;\n"
+               "    bool b = x;\n"
+               "}\n";
+        ASSERT_EQUALS(true, testValueOfX(code, 4U, 0));
+
+        code = "void f(int i) {\n"
+               "    if (i > 0) return;\n"
+               "    bool x = i < 0;\n"
+               "    bool b = x;\n"
+               "}\n";
+        ASSERT_EQUALS(true, testValueOfX(code, 4U, 1));
```

Comment:
Does your logic say that `x` is always 1 here? If `i==0` then `x` is 0.

Reply:
That is right. I dont think we will be able to fix the original FP without some kind of interval analysis. We should be able to fix some other FPs with this, though.

Reply:
What I wondered was if the `ValueFlow::Value` is known here? It should be possible here.